mlflow/sagemaker/__init__.py

```diff
@@ -845,3 +915,108 @@ def _delete_sagemaker_model(model_name, sage_client, s3_client):
     sage_client.delete_model(ModelName=model_name)
 
     return model_arn
+
+
+def _delete_sagemaker_endpoint_configuration(endpoint_config_name, sage_client):
+    """
+    :param sage_client: A boto3 client for SageMaker.
+    :return: ARN of the deleted endpoint configuration.
+    """
+    endpoint_config_info = sage_client.describe_endpoint_config(
+            EndpointConfigName=endpoint_config_name)
+    sage_client.delete_endpoint_config(EndpointConfigName=endpoint_config_name)
+    return endpoint_config_info["EndpointConfigArn"]
+
+
+def _find_endpoint(endpoint_name, sage_client):
+    """
+    Finds a SageMaker endpoint with the specified name in the caller's AWS account, returning a
+    NoneType if the endpoint is not found.
+
+    :param sage_client: A boto3 client for SageMaker.
+    :return: If the endpoint exists, a dictionary of endpoint attributes. If the endpoint does not
+             exist, `None`.
+    """
+    endpoints_page = sage_client.list_endpoints(
+        MaxResults=100, NameContains=endpoint_name)
+
+    while True:
+        for endpoint in endpoints_page["Endpoints"]:
+            if endpoint["EndpointName"] == endpoint_name:
+                return endpoint
+
+        if "NextToken" in endpoints_page:
+            endpoints_page = sage_client.list_endpoints(MaxResults=100,
+                                                        NextToken=endpoints_page["NextToken"],
+                                                        NameContains=endpoint_name)
+        else:
+            return None
+
+
+class _SageMakerOperation:
+
+    def __init__(self, status_check_fn, cleanup_fn):
+        self.status_check_fn = status_check_fn
+        self.cleanup_fn = cleanup_fn
+        self.start_time = datetime.now()
+        self.status = _SageMakerOperationStatus(_SageMakerOperationStatus.STATE_IN_PROGRESS, None)
+        self.cleaned_up = False
+
+    def await_completion(self, timeout_seconds):
+        begin = datetime.now()
```

Comment:
I would generally expect `time.time()` used here, since that's time relative to Unix epoch. Does datetime.now() have potential for daylight savings to move the time backwards?